backend/app/api/game_manager.py

```diff
@@ -172,4 +172,19 @@ def asdict(self) -> Dict[str, str]:
             'board':            self.board,
             'gameRunning':      self.game_running,
             'winner':           self.winner,
-        }
\ No newline at end of file
+        }
+
+    def one_random_letter(self, player : Player) -> None:
+        player.points -=  2
```

Comment:
what happens if a player has 0 points, do they go into negative points? If not, the function should probably raise some kind of error which will be caught and passed through to the frontend. I'm not 100% but I think a `ValueError` or some derivative should be raised.

Reply:
Yeah I definitely need to check this

Reply:
For the error, I was thinking of making another exception class for the perks and then raising it if the player used the perks when they weren't allowed to use it

Reply:
> For the error, I was thinking of making another exception class for the perks and then raising it if the player used the perks when they weren't allowed to use it

This would work fine as-well, in Python you can subclass exceptions. In this case I would subclass `ValueError`. To be honest though you probably don't need to unless you really want to. It's probably slightly overkill in this situation.

Reply:
So an alternative to creating my own exception, I could just raise the ValueError exception?

Reply:
> So an alternative to creating my own exception, I could just raise the ValueError exception?

Yeah I think that would probably be fine in this situation, if you look at the Python standard library or some other projects you'll see that they do create their own exception classes, I've actually done it a couple of times

```python
class InvalidWordException(Exception):
    """custom Exception class raised when the
    given word is invalid in Scrabble."""```

However, I actually might remove a couple of them since they're normally overkill and the inbuilt exceptions are fine. You may want to create your own custom exception if you think there's a possibility that another exception could be raised in the same function, i.e. you might have a function like this

```python
def parse_word(word: str):
    if len(word) == 0:
        raise ValueError("word cannot be an empty string")
   ... # do some parsing
   raise ValueError("parse failed")
```

in which case you might want to change the second error to a ParseError to make the distinction between the two errors.